src/Symfony/Component/Image/Image/Palette/Color/RGB.php

```diff
@@ -0,0 +1,214 @@
+                return $this->getGreen();
+            case ColorInterface::COLOR_BLUE:
+                return $this->getBlue();
+            default:
+                throw new InvalidArgumentException(sprintf('Color component %s is not valid', $component));
+        }
+    }
+
+    /**
+     * Returns RED value of the color
+     *
+     * @return integer
+     */
+    public function getRed()
+    {
+        return $this->r;
+    }
+
+    /**
+     * Returns GREEN value of the color
+     *
+     * @return integer
+     */
+    public function getGreen()
+    {
+        return $this->g;
+    }
+
+    /**
+     * Returns BLUE value of the color
+     *
+     * @return integer
+     */
+    public function getBlue()
+    {
+        return $this->b;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getPalette()
+    {
+        return $this->palette;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getAlpha()
+    {
+        return $this->alpha;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function dissolve($alpha)
+    {
+        return $this->palette->color(array($this->r, $this->g, $this->b), $this->alpha + $alpha);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function lighten($shade)
+    {
+        return $this->palette->color(
+            array(
+                min(255, $this->r + $shade),
+                min(255, $this->g + $shade),
+                min(255, $this->b + $shade),
+            ), $this->alpha
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function darken($shade)
+    {
+        return $this->palette->color(
+            array(
+                max(0, $this->r - $shade),
+                max(0, $this->g - $shade),
+                max(0, $this->b - $shade),
+            ), $this->alpha
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function grayscale()
+    {
+        $gray = min(255, round(0.299 * $this->getRed() + 0.114 * $this->getBlue() + 0.587 * $this->getGreen()));
+
+        return $this->palette->color(array($gray, $gray, $gray), $this->alpha);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isOpaque()
+    {
+        return 100 === $this->alpha;
+    }
+
+    /**
+     * Returns hex representation of the color
+     *
+     * @return string
+     */
+    public function __toString()
+    {
+        return sprintf('#%02x%02x%02x', $this->r, $this->g, $this->b);
+    }
+
+    /**
+     * Internal
+     *
+     * Performs checks for validity of given alpha value and sets it
+     *
+     * @param integer $alpha
+     *
+     * @throws InvalidArgumentException
+     */
+    private function setAlpha($alpha)
+    {
+        if (!is_int($alpha) || $alpha < 0 || $alpha > 100) {
+            throw new InvalidArgumentException(sprintf('Alpha must be an integer between 0 and 100, %s given', $alpha));
+        }
+
+        $this->alpha = $alpha;
+    }
+
+    /**
+     * Internal
```

Comment:
this short description must be removed, leaving ``Performs checks for color validity (array of array(R, G, B))`` as the actual short description